tests/containers/skopeo.pm

```diff
@@ -90,9 +92,44 @@ sub run {
     record_info('Compare images', 'Both extracted copies must be identical.');
     assert_script_run("diff -urN $dir1 $dir2", fail_message => 'Copied images are not identical.');
 
+    ######### Spin-up an instance of the latest Registry
+    assert_script_run("podman run --rm -d -p 5050:5000 --restart never --name skopeo-registry registry.suse.com/suse/registry:latest",
+        fail_message => "Failed to start local registry container");
+
+    ######### Pull the image into a our local repository
+    # skipping tls verification as by default most local registries don't have certificates
+    record_info('Copy Image', 'Copy image from remote repository into the local repository.');
+    validate_script_output("skopeo copy --dest-tls-verify=0 docker://$remote_image docker://$local_image",
+        sub { m/Writing manifest to image destination/ },
+        fail_message => 'Failed to copy image to local repository.');
+
+    ######### Inspect the local image repository
+    # skipping tls verification as by default most local registries don't have certificates
+    record_info('Inspect Image', 'Inspect an image from the local repository.');
+    assert_script_run("skopeo inspect --tls-verify=0 docker://$local_image",
+        fail_message => 'Failed to inspect local image.');
+
+    ######### Compare remote image to local image
+    # using JQ to omit any repository-specific fields which are expected to be different
+    record_info('Verify Remote Image', 'Inspect remote image and save results.');
+    assert_script_run("skopeo inspect --tls-verify=0 docker://$remote_image | jq .Layers[] >> $workdir/inspect_remote.json",
+        fail_message => 'Failed to inspect remote image.');
+
+    record_info('Verify Local Image', 'Inspect local image and save results.');
+    assert_script_run("skopeo inspect --tls-verify=0 docker://$local_image | jq .Layers[] >> $workdir/inspect_local.json",
```

Comment:
I think we can drop the `jq` usage by using the `-f` (`--format`) option with skopeo but feel free to ignore this suggestion.

Reply:
My idea with `jq` is to pull only the Layers, as other information can be modified by the registry. e.g. `Name` and `SHA` will be different due to the different registry. 

`--format` just allows me to specify json,yaml or other formatting. It does not have the ability to drop data afaik.

Reply:
> My idea with `jq` is to pull only the Layers, as other information can be modified by the registry. e.g. `Name` and `SHA` will be different due to the different registry.
> 
> `--format` just allows me to specify json,yaml or other formatting. It does not have the ability to drop data afaik.

It does, with `-f "{{ range .Layers }}{{ . }}\n{{ end }}"` but feel free to ignore since there's another instance of jq usage in this module.

Reply:
Are you sure that digests will be different because of the registry? That wouldn't sound like a proper way how to check image integrity then

Reply:
I assume the top-level `Digest` changes because the Registry itself is part of the formation of the SHA. e.g.

```
skopeo inspect docker://docker.io/library/alpine:latest
{
    "Name": "docker.io/library/alpine",
    "Digest": "sha256:0a4eaa0eecf5f8c050e5bba433f58c052be7587ee8af3e8b3910ef9ab5fbe9f5",
```

```
skopeo inspect --tls-verify=0 docker://localhost:5000/alpine:latest
{
    "Name": "localhost:5000/alpine",
    "Digest": "sha256:eddacbc7e24bf8799a4ed3cdcfa50d4b88a323695ad80f317b6629883b2c2a78",
```
This is the same image, copied with Skopeo. 

The `LayersData.Digest` however is identical on both.

IMO the Layers' SHA is a more trustworthy so I cherry-picked it as a test metric. At the end of the day, the main scope of the test is to verify that our version of Skopeo works with the latest Registry API.